frontend/lib/resolution/resolution-queries.cpp

```diff
@@ -2756,6 +2773,10 @@ static const Type* resolveBuiltinTypeCtor(Context* context,
     return t;
   }
 
+  if (auto t = getCPtrConstType(context, astForErr, ci)) {
+    return t;
+  }
+
```

Comment:
Since both `getCPtrType` and `getCPtrConstType` call the helper, this is completely redundant. If it returns `null` once, it will return `null` again. I suggest removing `getCPtrConstType` and `getCPtrType`, and renaming the helper back to `getCPtrType`, with the implication (and perhaps comment) that it handles the const case too.